### PR TITLE
Contact book

### PR DESCRIPTION
**Pull request by edumansky**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -459,6 +464,10 @@ function handleParentWindowMessage(event) {
     case "frameShown":
       handleFrameShownEvent();
       break;
+    case "listContacts":
+    console.log("LISTING CONTACTS");
```
Comment (edumansky) [inline src/App.js]: Get rid of these extra log lines

File: src/App.js [hunk 2]
```diff
@@ -467,6 +476,70 @@ function handleParentWindowMessage(event) {
...
+//send postMessage upon load of script
+//probably best to call this in contacts.js
+//add debugging in to figure out just how things are traveling
+//add finesse side of phonebook
+// finesse api? will have to figure out how user is created d
+function getContactsList(){
+  return contacts_list_global;
+}
+function formatNumbers(contact_list){
+  // starts with ( correct
```
Comment (edumansky) [inline src/App.js]: The logic here isn't right. First it ignores patterns like "5553331212" or "+15553331212", second we shouldn't assume that these are all 10 digit numbers and not, say, extension numbers for internal users.

We already have a google's phone formatting api included in the project so should use that. See `finesse_phone_api.js`, the library is `https://github.com/catamphetamine/libphonenumber-js`.

File: src/components/home/Contacts_helper.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    return `<li style="border-bottom: 1px solid #cccccc; background-color: #FFF; list-style-type: none; padding: 4px 0px;" id='${user.number}'>${userdiv} ${callNumber}</li>`
+  }
+
+  set_contact_list(contact_list){
+    this.contact_list = contact_list;
+  }
+
+  populate_contact_list(){
+    var target = document.getElementById("list_box");
```
Comment (edumansky) [inline src/components/home/Contacts_helper.js]: Redo the "manual" jquery html stuff as a React element

File: src/components/home/Contacts_helper.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    for(var i = 0; i < contact_list.length; i++){
+      if(contact_list[i].firstName.toString().toLowerCase().search(search_arg.toString().toLowerCase()) >= 0 || contact_list[i].lastName.toString().toLowerCase().search(search_arg.toString().toLowerCase()) >= 0 || contact_list[i].number.toString().toLowerCase().search(search_arg.toString().toLowerCase()) >= 0  ){
+        results.push(contact_list[i]);
+      }
+    }
+    var html = "";
+    try{
+      for(var i = 0; i < results.length; i++){
+          var user = results[i];
+          html += this.helper_cont.createUserhtml(user);
```
Comment (edumansky) [inline src/components/home/Contacts_helper.js]: There could potentially be many thousands of contacts in the unfiltered list, we don't want to just dump all of them into the html. Set some upper limit here, maybe display a little warning message at the bottom if limit is exceeded like "Only showing first X results"

File: src/components/home/Contacts_helper.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+import $ from "jquery";
+import Finesse from '../../finesse_apis/finesse_api';
+import FinessePhoneApi from '../../finesse_apis/finesse_phone_api';
+import ContactsView from './contacts'
+
+
+class Contacts_helper {
+
+  constructor(agent, phoneApi, tabNames) {
+    console.log("TESTING AGENT SHIT");
```
Comment (edumansky) [inline src/components/home/Contacts_helper.js]: Bruh

File: src/App.js [hunk 6]
```diff
@@ -467,6 +476,70 @@ function handleParentWindowMessage(event) {
...
+  // starts with number, remove first - and add ()
+  // starts with + add () and -
+  var returnlist = [];
+  var numberPattern = /\d+/g;
+  for(var i = 0; i < contact_list.length; i++){
+    var numbers = contact_list[i].number.match(numberPattern);
+    if(numbers == null){
```
Comment (edumansky) [inline src/App.js]: We need to support both business phone ("phone" field in SNOW) and mobile ("mobile_phone" field in SNOW). Bucher Suter's app does it with little phone icons, a desk phone icon for business phone and mobile phone icon for mobile. Maybe something like that if the text doesn't fit well. Either that or make the text smaller and have one on top of the other, with labels for "business:" and "mobile:"

File: src/App.js (hunk 1 above)
Comment (ryandavis709) [reply src/App.js]: fixed, commenting here so i can keep track of what i have left to do.

File: src/App.js (hunk 2 above)
Comment (ryandavis709) [reply src/App.js]: Fixed.

File: src/components/home/Contacts_helper.js (hunk 5 above)
Comment (ryandavis709) [reply src/components/home/Contacts_helper.js]: Lol whoops. Fixed

File: src/App.js (hunk 6 above)
Comment (ryandavis709) [reply src/App.js]: fixed

File: src/components/home/Contacts_helper.js (hunk 3 above)
Comment (ryandavis709) [reply src/components/home/Contacts_helper.js]: fixed.

File: src/components/home/Contacts_helper.js (hunk 4 above)
Comment (ryandavis709) [reply src/components/home/Contacts_helper.js]: fixed, currently limited to 15 contacts showing

Comment (ryandavis709) [reply src/components/home/Contacts_helper.js]: updated to 100 contacts, will change into a setting from openframe

**Pull request merged**